indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,296 @@
+
+import {
+  FinalCheckbox,
+  FinalRadio,
+  FinalField,
+  FinalInput,
+  validators as v,
+} from 'indico/react/forms';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Translate, PluralTranslate} from 'indico/react/i18n';
+import {camelizeKeys} from 'indico/utils/case';
+import {$T} from 'indico/utils/i18n';
+
+import '../../../styles/regform.module.scss';
+import {getStaticData} from '../selectors';
+
+function TimetableSessionsComponent({
+  value,
+  name,
+  sessionData,
+  onChange,
+  onFocus,
+  onBlur,
+  allowFullDays,
+  display,
+  minimum,
+  maximum,
+  ...props
+}) {
+  const markTouched = () => {
+    onFocus();
+    onBlur();
+  };
+
+  function handleValueChange(e, data) {
+    markTouched();
+    const id = data.value;
+    const newValue = value.includes(id) ? value.filter(el => el !== id) : [...value, id];
+    onChange(newValue);
+  }
+
+  function formatDate(date) {
+    const options = {
+      year: 'numeric',
+      month: 'short',
+      day: 'numeric',
+      weekday: 'short',
+    };
+    const dt = new Date(date);
+    return dt.toLocaleDateString(undefined, options);
+  }
+
+  const currentSelectionMsg = $T.gettext('You selected {0} session blocks.').format(value.length);
+
+  return (
+    <Form.Group styleName="timetablesessions-field">
+      <strong>{currentSelectionMsg}</strong>
+      {sessionData &&
+        Object.keys(sessionData)?.map((date, index) => (
+          <SessionBlockHeader
+            value={value}
+            index={index}
+            name={name}
+            data={sessionData[date]}
+            label={formatDate(date)}
+            allowFullDays={allowFullDays}
+            display={display}
+            key={`session-block-${date}`}
+            handleValueChange={handleValueChange}
+            onChange={onChange}
+            props={props}
+          />
+        ))}
+    </Form.Group>
+  );
+}
+
+TimetableSessionsComponent.propTypes = {
+  value: PropTypes.arrayOf(PropTypes.number).isRequired,
+  name: PropTypes.string.isRequired,
+  sessionData: PropTypes.objectOf(
+    PropTypes.arrayOf(
+      PropTypes.shape({
+        full_title: PropTypes.string.isRequired,
+        id: PropTypes.number.isRequired,
+        start_dt: PropTypes.string.isRequired,
+      })
+    )
+  ),
+  onChange: PropTypes.func.isRequired,
+  onFocus: PropTypes.func.isRequired,
+  onBlur: PropTypes.func.isRequired,
+  allowFullDays: PropTypes.bool.isRequired,
+  display: PropTypes.oneOf(['expanded', 'collapsed']),
+  maximum: PropTypes.number.isRequired,
+  minimum: PropTypes.number.isRequired,
+};
+
+TimetableSessionsComponent.defaultProps = {
+  sessionData: {},
+};
+
+function SessionBlockHeader({
+  value,
+  data,
+  label,
+  allowFullDays,
+  display,
+  handleValueChange,
+  onChange,
+}) {
+  const childrenIds = data.map(e => e.id);
+  const isChecked = childrenIds.reduce((acum, id) => {
+    return acum && value.includes(id);
+  }, true);
```

Comment:
```suggestion
  const isChecked = childrenIds.every(id => value.includes(id));
```
Not every efficient though, converting `value` to a `Set` would probably be better (also because there's lots of `.includes` calls)